### PR TITLE
Revert "Setup Java 21 container (#28833)"

### PR DESCRIPTION
**Pull request by damccorm**
This reverts commit e8e38145d96a7b0561a418c19519917b0567591e.

Looks like this is causing publishing snapshots to fail - https://ci-beam.apache.org/job/beam_Publish_Beam_SDK_Snapshots/6269/console

This might be a problem elsewhere (and probably is on GHA, though that's currently busted too - https://github.com/apache/beam/actions/workflows/beam_Publish_Beam_SDK_Snapshots.yml)

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @Abacn

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Comment by Abacn** [comment]
I see, that's because Jenkins does not have Java21 installed yet, yes we should revert

File: .test-infra/jenkins/JavaTestProperties.groovy [hunk 1]
```diff
@@ -17,10 +17,5 @@
  */
 
 class JavaTestProperties {
-  final static List<String> SUPPORTED_CONTAINER_TASKS = [
-    'java8',
-    'java11',
-    'java17',
-    'java21'
```
Comment (Abacn) [inline .test-infra/jenkins/JavaTestProperties.groovy:24]: this line causes Jenkins trying to build Java21 container as part of publish SDK snapshot job

**Comment by damccorm** [comment]
Run SQL PreCommit

**Comment by damccorm** [comment]
Run Java_Kinesis_IO_Direct PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

**Comment by damccorm** [comment]
Run Java PreCommit

**Comment by Abacn** [comment]
actually, there is another cause is that the build sdk snapshot task currently always set `-Pjava11Home` even for Java17. 

https://github.com/apache/beam/blob/71b8e93b6d385d1117e2659ee3f362ccec3422f6/.test-infra/jenkins/job_Publish_SDK_Image_Snapshots.groovy#L53

This worked by coincidence and start failing after the change of #28833

**Comment by Abacn** [comment]
merge for now and will provide a roll forward PR that fixes these issues

**Review by Abacn — APPROVED** [review]

**Comment by Abacn** [comment]
run seed job

**Comment by Abacn** [comment]
all test passed (except codecov diff report which is not relevant here)

**Pull request merged**